validation/validation.go

```diff
@@ -0,0 +1,370 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+)
+
+// New is the main function in charge of validating the HTTP request form by using the defined rule set.
+// Form fields will be validated only if there is a rule that indicates they must be validated.
+func New(form url.Values, ruleSet ...Rule) map[string][]string {
+	verrs := make(map[string][]string)
+	mutex := new(sync.RWMutex)
+
+	for _, rule := range ruleSet {
+		mutex.Lock()
+		verrs[rule.Field] = append(verrs[rule.Field], rule.validate(form[rule.Field]...)...)
+		mutex.Unlock()
+	}
+
+	for k, v := range verrs {
+		if len(v) == 0 {
+			delete(verrs, k)
+		}
+	}
+
+	return verrs
+}
+
+// Validation is a condition that must be satisfied by all values in a specific form field.
+// or else an error message is displayed indicating that at least one value is invalid.
+type Validation func(...string) error
```

Comment:
The signature of this type should have both, bool and error so that it can tell if its valid or signal there was an error in checking when there is an external condition to do the check p.e. database.

Reply:
I agree, we need a way to handle any kind of issue when the validation could be performed correctly.

Reply:
I wonder if signature can better return a string pointer and an error. So in case that any value is invalid, the pointer can be addressed with a validation error message